# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1452271101 18000
#      Fri Jan 08 11:38:21 2016 -0500
# Node ID 8190fe6dc3fd116efbb12bc53554e52b7614934f
# Parent  43b498b8c38cee239fce988ea67bf31f36e1d5ad
* data.cc (Fcumsum, Fcumprod): Provide example in docstring.

diff --git a/libinterp/corefcn/data.cc b/libinterp/corefcn/data.cc
--- a/libinterp/corefcn/data.cc
+++ b/libinterp/corefcn/data.cc
@@ -1152,16 +1152,27 @@ negative numbers or when the values are 
 
 DEFUN (cumprod, args, ,
        "-*- texinfo -*-\n\
 @deftypefn  {} {} cumprod (@var{x})\n\
 @deftypefnx {} {} cumprod (@var{x}, @var{dim})\n\
 Cumulative product of elements along dimension @var{dim}.\n\
 \n\
 If @var{dim} is omitted, it defaults to the first non-singleton dimension.\n\
+For example:\n\
+\n\
+@example\n\
+@group\n\
+cumsum ([1, 2; 3, 4; 5, 6])\n\
+   @result{}  1   2\n\
+       3   8\n\
+      15  48\n\
+@end group\n\
+@end example\n\
+@end example\n\
 @seealso{prod, cumsum}\n\
 @end deftypefn")
 {
   DATA_REDUCTION (cumprod);
 }
 
 /*
 %!assert (cumprod ([1, 2, 3]), [1, 2, 6])
@@ -1188,16 +1199,27 @@ DEFUN (cumsum, args, ,
 @deftypefn  {} {} cumsum (@var{x})\n\
 @deftypefnx {} {} cumsum (@var{x}, @var{dim})\n\
 @deftypefnx {} {} cumsum (@dots{}, \"native\")\n\
 @deftypefnx {} {} cumsum (@dots{}, \"double\")\n\
 @deftypefnx {} {} cumsum (@dots{}, \"extra\")\n\
 Cumulative sum of elements along dimension @var{dim}.\n\
 \n\
 If @var{dim} is omitted, it defaults to the first non-singleton dimension.\n\
+For example:\n\
+\n\
+@example\n\
+@group\n\
+cumsum ([1, 2; 3, 4; 5, 6])\n\
+   @result{}  1   2\n\
+       4   6\n\
+       9  12\n\
+@end group\n\
+@end example\n\
+@end example\n\
 \n\
 See @code{sum} for an explanation of the optional parameters\n\
 @qcode{\"native\"}, @qcode{\"double\"}, and @qcode{\"extra\"}.\n\
 @seealso{sum, cumprod}\n\
 @end deftypefn")
 {
   int nargin = args.length ();
 
